### PR TITLE
Reorganize sphinx doc

### PR DESCRIPTION
**Pull request by Quasimewdo**
A documentation is essential for user to learn the API, this
patch improves the organization of doc.

### CODE REVIEW

**Comment by yihong0618** [comment]
Amazing work!

File: greenplumpython/type.py [hunk 1]
```diff
@@ -25,14 +95,17 @@ def create_type(
     is_temp: bool = True,
 ) -> str:
     """
-    Returns:
-        Name of created composite type
+    Creates a new composite type in database and return its name
```
Comment (higuoxing) [inline greenplumpython/type.py:98]:
```suggestion
    Creates a new composite type in database and returns its name
```

File: greenplumpython/expr.py [hunk 2]
```diff
@@ -293,61 +412,11 @@ def _serialize(self) -> str:
...
-
-    return Type(name, db=db)
-
-
 class Column(Expr):
     """
-    Inherited from Expr. Representation of a python object Column.
+    Inherited from Expr.
+
+    Representation of a python object Column.
```
Comment (higuoxing) [inline greenplumpython/expr.py:419]:
```suggestion
    Representation of a Python object Column.
```

File: greenplumpython/table.py [hunk 3]
```diff
@@ -499,6 +541,12 @@ def fetch(self, is_all: bool = True) -> Iterable[Tuple[Any]]:
         Fetch rows of this table.
         - if is_all is True, fetch all rows at once
         - otherwise, open a CURSOR and FETCH one row at a time
+
+        Args:
+            is_all: bool: Define if fetch all rows at once
+
+        Returns:
+            Iterable[Tuple[ANy]]: results of query received from database
```
Comment (higuoxing) [inline greenplumpython/table.py:549]:
```suggestion
            Iterable[Tuple[Any]]: results of query received from database
```

File: greenplumpython/type.py [hunk 4]
```diff
@@ -25,14 +95,17 @@ def create_type(
...
 
     Args:
         class_type : object : class which user want to reproduce in Greenplum
         db : Database : where the type will be created
         as_name : Optional[str] : name of the created type if different from class
         is_temp : bool : if type exists only for current session
+
+    Returns:
+        str: name of created composite type
```
Comment (higuoxing) [inline greenplumpython/type.py:107]:
```suggestion
        str: name of the created composite type
```

**Review by higuoxing — APPROVED** [review]
Great work! LGTM with some minor suggestions.

File: doc/source/group.rst [hunk 5]
```diff
@@ -3,4 +3,6 @@ Grouping
 
 .. automodule:: group
    :members:
+   :special-members: __or__
+   :exclude-members: make_group_by_clause, get_targets
```
[the remark this answers is not in the document]
Comment (Quasimewdo) [reply doc/source/group.rst:7]: `:meta private:` works, but `:meta public:` doesn't work to show ` __or__` in docstring

File: greenplumpython/expr.py (hunk 2 above)
Comment (Quasimewdo) [reply greenplumpython/expr.py:419]: Which means?

Comment (Quasimewdo) [reply greenplumpython/expr.py]: Do you mean like `Representation of a Python object :class:`.Column`.`

**Pull request merged**